database/migrations/2014_10_12_000001_modify_users_table.php

```diff
@@ -16,12 +16,21 @@ public function up()
         });
 
         Schema::table('users', function (Blueprint $table) {
+            // @todo need better testing solution
             $column = $table->string('last_name')->after('first_name');
             // Workaround for Sqlite / ALTER
             // - https://stackoverflow.com/questions/20822159/laravel-migration-with-sqlite-cannot-add-a-not-null-column-with-default-value-n
             if (config("database.default") === "testing") {
                 $column->default('');
             }
+            // Temporary workaround for Sqlite / ALTER
+            // @todo need better testing solution
+            if (config("database.default") !== "testing") {
+                $table->string('username')->unique()->after('last_name');
+            }
+            if (config("database.default") === "testing") {
+                $table->string('username')->nullable()->after('last_name');
+            }
```

Comment:
@jasonmccreary Could you take a look at this and recommend a better approach for sqlite testing with this issue of a migration modifying the user table?

Reply:
To my knowledge, `after()` should have no affect for SQLite. It only generates SQL when the underlying driver is MySQL.

I use `after` in some migrations and run SQLite for local testing without problem. What was the original error that led to this conditional wrapping?

Reply:
@drewroberts ☝️ (forgot to tag you)

Reply:
@jasonmccreary Here is what the error said:

`SQLSTATE[HY000]: General error: 1 Cannot add a NOT NULL column with default value NULL`

You can view it here:

- https://github.com/tipoff/authorization/runs/2029922988

Reply:
@drewroberts that seems more to do with a new column not being `nullable` rather than using `after`.

When data exists in the database and you add a new column, it must be either nullable or set a default value so existing records are properly updated as well.

Reply:
OK, that makes sense. It seemed to me that we couldn't do it in a migration that modifies a table. I'll open a new PR and tag you on it so you can take a look at fixing it.